OpenSim/Analyses/OutputReporter.h

```diff
@@ -0,0 +1,102 @@
+#ifndef OUTPUT_REPORTER_H_
+#define OUTPUT_REPORTER_H_
+/* -------------------------------------------------------------------------- *
+ *                         OpenSim: OutputReporter.h                          *
+ * -------------------------------------------------------------------------- *
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2017 Stanford University and the Authors                *
+ * Author(s):                                                                 *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+
+//=============================================================================
+// INCLUDES
+//=============================================================================
+#include <OpenSim/Simulation/Model/Analysis.h>
+#include <OpenSim/Common/Reporter.h>
+#include <OpenSim/Common/STOFileAdapter.h>
+#include "osimAnalysesDLL.h"
+
+#ifdef SWIG
+    #ifdef OSIMANALYSES_API
+        #undef OSIMANALYSES_API
+        #define OSIMANALYSES_API
+    #endif
+#endif
+//=============================================================================
+//=============================================================================
+namespace OpenSim { 
+
+/**
+ * The OutputReporter Analysis is a wrapper for a TableReporter. It generates
+ * and writes TimeSeriesTables of Output values according to the names listed
+ * as properties of the Analysis. OutputReporter enables the AnalyzeTool to
+ * report on Outputs via the Analysis interface but backed by a TableReporter.
+ * The OutputReporter currently only supports Outputs of type, double, Vec3
+ * and SpatialVec.
```

Comment:
Please mention exactly what files are written, and how it depends on the types of the outputs the user lists.